sync-team/src/github/mod.rs

```diff
@@ -1027,9 +1005,7 @@ enum EnvironmentDiff {
         new_branches: Vec<String>,
```

Comment:
I think that it would be better if the sync code directly operated on branches/tags to add or remove, rather than having generic code that then tries to load the current state from GitHub again when doing the sync.

In other words, I would remove `new_branches` and `new_tags`, and keep only add/remove.